### PR TITLE
[WasmFS][NFC] Moved ParsedPath to paths.h

### PR DESCRIPTION
**Pull request by tlively**
This was left out of #16286 by accident.

### CODE REVIEW

**Review by kripken — APPROVED** [review]

File: system/lib/wasmfs/paths.h [hunk 1]
```diff
@@ -11,6 +11,11 @@
 
 namespace wasmfs {
 
+struct ParsedPath {
+  std::optional<Directory::Handle> parent;
```
Comment (sbc100) [inline system/lib/wasmfs/paths.h:15]: Surely all files have a parent.. even those that live in the root ?

Comment (tlively) [reply system/lib/wasmfs/paths.h:15]: Not files that have been unlinked and are only kept alive by the open file table!

Comment (kripken) [reply system/lib/wasmfs/paths.h:15]: Also the root itself has no parent.

Comment (tlively) [reply system/lib/wasmfs/paths.h:15]: Oh yeah, I guess an unlinked file wouldn't be returned by path parsing anyway.

Comment (kripken) [reply system/lib/wasmfs/paths.h:15]: Hmm, it could be unlinked by another thread after parsing, though...

**Pull request merged**